protocol/x/clob/keeper/orders.go

```diff
@@ -130,18 +131,36 @@ func (k Keeper) PlaceShortTermOrder(
 func (k Keeper) CancelStatefulOrder(
 	ctx sdk.Context,
 	msg *types.MsgCancelOrder,
-) error {
+) (err error) {
+	defer func() {
+		if err != nil {
+			callback := metrics.DeliverTx
+			if ctx.IsCheckTx() {
+				callback = metrics.CheckTx
+			}
+			if ctx.IsReCheckTx() {
+				callback = metrics.ReCheckTx
+			}
```

Comment:
nit: I assume we are going to do this a lot in the future . let's move this to a helper function

Reply:
Yeah maybe a `GetCallback` function

Also should note that we assume if it's not `CheckTx` and `RecheckTx` that it's `DeliverTx`, which isn't always true